pyk/src/pyk/prelude.py

```diff
@@ -68,15 +68,14 @@ def build_assoc(unit: KInner, label: Union[str, KLabel], terms: Iterable[KInner]
     return res or unit
 
 
-def buildCons(unit, cons, ls):
-    """Build a cons operator term given the cons and unit ops.
-
-    -   Input: unit, cons, and list of elements to join.
-    -   Output: cons-list style construction of the joined term.
-    """
-    if len(ls) == 0:
+def build_cons(unit: KInner, label: Union[str, KLabel], terms: Iterable[KInner]) -> KInner:
+    label = label if type(label) is KLabel else KLabel(label)
+    it = iter(terms)
```

Comment:
Is it necassary to process `label` like this? It's being used directly in `KApply`, which already does the processing.

maybe we should make a new type for the type `Union[str, KLabel]`, or can we somehow automatically lift type `str` to type `KLabel`?

Reply:
The better typing can be dealt with in another PR.

Reply:
> Is it necassary to process `label` like this?

Good point, fixed.

> can we somehow automatically lift type str to type KLabel?

We can write a helper function for `label if type(label) is KLabel else KLabel(label)`, but I think that's it.